### DISCUSSION TITLE
Arbitrary Runner support?

### DISCUSSION THREAD

**Discussion by wom** [LazyVim/LazyVim#1463] [Ideas]
Hey, experimenting with migrating to LazyVim. 

In my existing (not lazy.vim config, I have a snippet that uses Overseer to run 'current file' and display results in quickfix.  I've had similar snippets in vim for ~15 years - would similar functionality make sense to bake directly into LazyVim?  I find it invaluable for quick prototyping smaller scripts (verify a quick bash expansion, python regex, etc), and have had some variation of this in my ~/.vimrc for ~15 years now.  

I'm not suggesting this exact implementation, but throwing it out there as something I see as a valuable default mapping in LazyVim via (some method).  I see a few other more language-specific discussions, but I do think this could cover most prototyping wants.

```
local overseer = require("overseer")
overseer.setup()
overseer.register_template({
  name = "Runner",
  builder = function(params)
    return {
        cmd = {
            vim.api.nvim_buf_get_name(0),
        },
        components = {
            "on_output_quickfix",
            "default",
            "on_complete_notify",
        },
    }
  end,
  tags = { overseer.TAG.BUILD },
  params = {},
  priority = 4,
  condition = {
    filetype = { "py", "sh", "lua", "go" },
  },
})

local vkms = vim.keymap.set
vkms("n", "<Leader>e", function()
    local overseer = require("overseer")
    overseer.run_template({name = "Runner"}, function(task)
        if task then
            overseer.run_action(task, '')
            -- Task writes to quickfix list by default, so open it.
            vim.cmd("copen")
        end
    end)
end)
```

**Comment by meicale** [comment]
That is great, I am using overseer.nvim and other plugins (such as "Zeioth/compiler.nvim", "pianocomposer321/officer.nvim",  "meicale/overseer-quick-tasks",
...) base on it. It is a great the vim-way runner and It is much more convenient. I also suggest that make it in extra at least, I think people need will love that!
